packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts

```diff
@@ -111,85 +149,147 @@ class InteractiveGraphQuestionBuilder {
         return this;
     }
 
-    withSegments(
-        startCoords: CollinearTuple[],
-    ): InteractiveGraphQuestionBuilder {
-        this.interactiveFigureConfig = new SegmentGraphConfig(
-            startCoords.length,
-            startCoords,
-        );
+    withSegments(options?: {
+        numSegments?: number;
```

Comment:
It might be worth having a comment here to say that if `numSegments` is not passed, it gets defaulted appropriately.

Alternatively, I might be inclined to split `withSegments` into multiple methods for different use cases:

```
withDefaultSegments(numSegments)

withSegments({startCoords, coords})
```

Reply:
(talked about offline) I think for now, I'd prefer to leave it as one method so it's consistent with the other graph type methods. Even for the `withSegments` example, there's a potential mismatch between the length of `startCoords` and the length of `coords`. Ideally, we'd be able to trust the devs. What I really think would happen is that if they needed to test the length, it would fail.

If it becomes an issue, we can change it later. For now, I found this way to be the most intuitive to me while I was trying to use this method in the test data.